agent/ec2/ec2_metadata_client.go

```diff
@@ -138,6 +140,11 @@ func (c *ec2MetadataClientImpl) PrimaryENIMAC() (string, error) {
 	return c.client.GetMetadata(MacResource)
 }
 
+// AllENIMacs returns the mac addresses for all the network interfaces attached to the instance
```

Comment:
just curious, why not a list of strings? Plural method name, returning a delimited single string struck me odd.

Reply:
this is a wrapper method that simply uses ec2's [GetMetadata](https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/#EC2Metadata.GetMetadata) API and returns the result as it. the result is converted to a list of strings in the [GetNetworkInterfaceMacs](https://github.com/fenxiong/amazon-ecs-agent/blob/eni-trunking-func/agent/functional_tests/util/utils.go#L975-L980) method